### ISSUE TITLE
Error:  Failed to copy ConEmu.xml file to user-conemu.xml backup location! 

### ISSUE THREAD

**Issue by SmGG** [cmderdev/cmder#1930]
When I use vs Code debug and set the terminal to cmder, I got this Error 
> Failed to copy ConEmu.xml file to user-conemu.xml backup location! 

this is the configuration in vs Code 
> "terminal.external.windowsExec": "C:\\Users\\Smile_Gum_Gum\\AppData\\Local\\cmder\\Cmder.exe",

Can anybody help me out

**Comment by daxgames** [comment]
`Cmder.exe` is a dumb launcher that does the dile copy that is failing.  `init.bat` does all the work.

I am not a VS Code user but many of our users are and this is how they integrate cmder, I assume for the same purpose you are trying to do.  @drsdavidsoft can you look at this and confirm?

**Comment by daxgames** [comment]
@SmGG I am not sure if the below would work but it looks to me like this might achieve what you want **ASSUMING** it is valid:

_Note: Substitute [cmder_root] with your Cmder installation directory._

```
"terminal.external.windowsExec": "cmd.exe",
"terminal.external.windowsExecArgs": [
      "/k", "[cmder_root]\\vendor\\init.bat"
],
```

**Comment by DRSDavidSoft** [comment]
@SmGG & @daxgames  – Sorry for the late response.

I'd like to clear some up some information on the way both Cmder and VS Code. Hopefully this can also be used a future reference for similar problems.

**TL;DR:** make sure you are on the latest version and also whether you get the same error message if you get the same error message when you launch the terminal directly (and not using the debug menu)

<br />

I decided to write a complete explanation on how Cmder and VS Code work internally for future references.

## Cmder is a software package
A rather important thing to grasp on is that _Cmder is not a stand-alone executable_, but a collection of different software items that are packaged together in order to provide a complete terminal.

For the sake of simplicity, I break up the contents to the following:
1. The console (terminal window)
2. The internal shell

Whenever _Cmder.exe_ is launched, a new **console** window is created and loaded with the Cmder **shell**. 

<img height="48" src="https://user-images.githubusercontent.com/4673812/49048396-d05c7c80-f1ef-11e8-8e76-2fb6d81b6e79.png">

This is the typical window you see when you double-click on the _Cmder.exe_ binary:

<img src="https://conemu.github.io/img/cmder.png" />

Cmder is packaged with [ConEmu](https://conemu.github.io/) as the console window, and uses [clink](https://mridgers.github.io/clink/) with some special configuration as the default shell.  The window you see in the above photo, is actually the packaged ConEmu that _emulates the terminal_ in a window.

As @daxgames said, the _Cmder.exe_ binary is just a launcher that opens the ConEmu with the Cmder shell.  (The main core of Cmder is a batch file located at `vendor/init.bat` which provides the actual shell.)

<br>

## How the terminal in VS Code works
There are actually **two** different modes that the VS Code terminal operates in:
1. `integrated`: The terminal is rendered _inside_ the VS Code's window _– (a.k.a the "Seamless mode" here)_
2. `external`: VS Code launches an external terminal.

### How do they differ?
* In the first mode, the Cmder's **shell** (consisting of clink and all the other goodies) are ran _inside a frame_ in the VS Code window, hence the name "integrated terminal". This is referred to as the "Seamless mode" in the Cmder wiki and guides. (The default hotkey is <kbd>Ctrl</kbd> + <kbd>~</kbd>.)
* As the opposite, the "external terminal" mode will simply launch a new terminal window. (The default hotkey is <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>C</kbd>.)

**Here's a demo of the first mode:**
<img height="156" src="https://i.stack.imgur.com/0QnHy.png">

**Here's the relevant configuration for both methods:**
```json
    "terminal.integrated.shell.windows": "cmd.exe",
    "terminal.integrated.shellArgs.windows": [
        "/k", "%CMDER_ROOT%\\vendor\\init.bat"
    ],

    "terminal.external.windowsExec": "C:\\path\\to\\cmder\\Cmder.exe",
```

Now I'm not quite sure as to why @SmGG would like to use the external window method instead of the integrated method, but the above configuration should work just fine in this case.

### There are a couple things I need to mention:
1. **Make sure you're using the latest version of Cmder.** The code for `Cmder.exe` launcher is always being updated, and your issue might be fixed if you update. [How to see which version you are using?](https://github.com/cmderdev/cmder/wiki/Cmder-troubleshooting#how-do-i-check-which-version-of-cmder-i-am-using)

2. **Test running Cmder as the external terminal without using the debug option. (<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>C</kbd>)** You should not get the following message:
    > _Failed to copy ConEmu.xml file to user-conemu.xml backup location!_

    * If you are getting this message **without using the debug menu**, there is something wrong with the launcher code that we need to fix.
    * If this error message **only appears with the debug menu**, we need to consider adding this as a new functionality.

3. **Always use the appropriate configuration for each method:**
    * the _integrated terminal_ method should only call the `init.bat` file to adjust the shell.
    * the _external terminal_ should always call a **terminal emulator** (such as the ConEmu via _Cmder.exe_) to run.

4. @daxgames, the `"terminal.external.windowsExecArgs"` thing that you suggested is _not_ a valid VS Code settings option, and does not have any effect.

**Comment by daxgames** [comment]
@DRSDavidSoft Thanks for the write-up.  I am not a VSCode user so I was guessing.  That is why I asked you to chime in since I know you could help.

**Comment by DRSDavidSoft** [comment]
@daxgames I'm always glad to be of any help! Let's see if this guide is useful in this case.
